FROST-Server.SQLjooq/src/main/resources/liquibase/postgresTriggers.sql

```diff
@@ -186,40 +276,80 @@ drop trigger if exists datastreams_actualization_delete ON "OBSERVATIONS";
 
 -- ---------------------------------------
 -- Function: datastreams_update_delete()
+--
+-- This function also updates multidatastreams. Updated fields are:
+-- PHENOMENON_TIME_START,PHENOMENON_TIME_END,RESULT_TIME_START,RESULT_TIME_END.
+-- Warning: OBSERVED_AREA not taken into account. 
 -- ---------------------------------------
 create or replace function datastreams_update_delete()
   returns trigger as
 $BODY$
 declare
 "DS_ROW" "DATASTREAMS"%rowtype;
+"MDS_ROW" "MULTI_DATASTREAMS"%rowtype;
 begin
 
-for "DS_ROW" in select * from "DATASTREAMS" where "ID"=OLD."DATASTREAM_ID"
-loop
-    if (OLD."PHENOMENON_TIME_START" = "DS_ROW"."PHENOMENON_TIME_START"
-        or coalesce(OLD."PHENOMENON_TIME_END", OLD."PHENOMENON_TIME_START") = "DS_ROW"."PHENOMENON_TIME_END")
-    then
-        update "DATASTREAMS"
-            set "PHENOMENON_TIME_START" = (select min("PHENOMENON_TIME_START") from "OBSERVATIONS" where "OBSERVATIONS"."DATASTREAM_ID" = "DS_ROW"."ID")
-            where "DATASTREAMS"."ID" = "DS_ROW"."ID";
-        update "DATASTREAMS"
-            set "PHENOMENON_TIME_END" = (select max(coalesce("PHENOMENON_TIME_END", "PHENOMENON_TIME_START")) from "OBSERVATIONS" where "OBSERVATIONS"."DATASTREAM_ID" = "DS_ROW"."ID")
-            where "DATASTREAMS"."ID" = "DS_ROW"."ID";
-    end if;
-
-    if (OLD."RESULT_TIME" = "DS_ROW"."RESULT_TIME_START")
-    then
-        update "DATASTREAMS"
-            set "RESULT_TIME_START" = (select min("RESULT_TIME") from "OBSERVATIONS" where "OBSERVATIONS"."DATASTREAM_ID" = "DS_ROW"."ID")
-            where "DATASTREAMS"."ID" = "DS_ROW"."ID";
-    end if;
-    if (OLD."RESULT_TIME" = "DS_ROW"."RESULT_TIME_END")
-    then
-        update "DATASTREAMS"
-            set "RESULT_TIME_END" = (select max("RESULT_TIME") from "OBSERVATIONS" where "OBSERVATIONS"."DATASTREAM_ID" = "DS_ROW"."ID")
-            where "DATASTREAMS"."ID" = "DS_ROW"."ID";
-    end if;
-end loop;
+if (OLD."DATASTREAM_ID" is not null) 
+then
+	select * into "MDS_ROW" from "MULTI_DATASTREAMS" where "MULTI_DATASTREAMS"."ID"=OLD."MULTI_DATASTREAM_ID";
```

Comment:
Shouldn't that be 
`select * into "DS_ROW" ...`
?

Reply:
Yes, you're right: it was and it still  should be DS_ROW. Thanks.

Reply:
line should have remained unchanged:
`select * into "DS_ROW" from DATASTREAMS" where DATASTREAMS"."ID"=OLD."DATASTREAM_ID";`